pytket/extensions/quantinuum/backends/quantinuum.py

```diff
@@ -366,6 +366,24 @@ def default_compilation_pass(self, optimisation_level: int = 2) -> BasePass:
         ]
         squash = auto_squash_pass({OpType.PhasedX, OpType.Rz})
 
+        # In TKET, a qubit register with N qubits can have qubits
+        # indexed with a a value greater than N, i.e. a single
+        # qubit register can exist with index "7" or similar.
+        # Similarly, a qubit register with N qubits could be defined
+        # in a Circuit, but fewer than N qubits in the register
+        # have operations.
+        # Both of these cases can causes issues when converting to QASM,
+        # as the size of the defined "qreg" can be larger than the
+        # number of Qubits actually used, or at times larger than the
+        # number of device Qubits, even if fewer are really used.
+        # By flattening the Circuit qubit registers, we make sure
+        # that the produced QASM has one "qreg", with the exact number
+        # of qubits actually used in the Circuit.
+        def flatten_registers(c: "Circuit") -> "Circuit":
+            c.remove_blank_wires()
+            c.rename_units({c.qubits[i]: Qubit(i) for i in range(len(c.qubits))})
```

Comment:
My only concern is that the renaming map is not visible to the user. Since we are not returning the map to the user maybe we should add a note explain how the renaming is done (i.e. c.qubits[i] -> Qubit(i))?

Reply:
Happy to add a note explaining this. The reason I haven't is that we also don't provide that information when we map to a physical Circuit in other devices. I'll also change from `Qubit(i)` to `Qubit("quantinuum", i)`  to make this clearer.

Reply:
Thanks, I think that should be sufficient. 

In other devices, shouldn't the user still be able to know the placement by wrapping the circuit in a `CompilationUnit` and compile the unit with the `get_default_compilation_pass` ? I might be completely wrong.

Reply:
Ah ok, yes they can do that. So the concern is that `CustomPass` might not update the `maps` attribute suitably? Maybe we can raise a JIRA ticket to add a flattening pass to TKET for a followup?

Reply:
Sounds good!